api/query/cache/index/aggregator.go

```diff
@@ -54,7 +61,7 @@ func (a *indexAggregator) Add(t TestID) error {
 
 		if a.includeSubtests {
 			if _, name, err := ts.GetName(t); err == nil && name != nil {
-				r.Subtests = append(r.Subtests, *name)
+				r.Subtests.Add(*name)
```

Comment:
why do we need a set to store pointers? to make sure no pointers that point to the same memory address?

Reply:
`*name` is dereferencing the pointer`s value (string).
In most cases, the same subtests exist for all of the runs. We're taking the actual string values in the set, so that we don't have duplication.